xformers/triton/patch.py

```diff
@@ -0,0 +1,41 @@
+import typing
+
+import torch
+import torch.nn
+import torch.nn.functional
+
+from dropout import dropout
+from fused_linear_layer import FusedLinear
+from layer_norm import FusedLayerNorm
+from softmax import softmax, log_softmax
+
+
+def _softmax(input: torch.Tensor, dim: int = -1, dtype=torch.float32):
+    if dim == -1:
+        return softmax(input)
```

Comment:
Should this be limited to triton? For example there is sparse softmax as well available [here](https://github.com/facebookresearch/xformers/blob/861493c509de8bfc68895c9b5c58fbbc901d5176/xformers/components/attention/core.py#L96)

Reply:
I'm not sure I understand. Do you want to call the `_softmax` function instead? I wrote this wrapper to add transpose(), which we don't need with PyTorch softmax.

Reply:
Seems like you're replacing `torch.nn.functional.softmax` with triton softmax only. Just suggesting you may want to also patch with sparse softmax if using sparse tensors or have a fallback when triton is not available, unless the focus of this is just triton? Then the way it is now seems fine.